src/generators/samples/rlcSampleGenerator.ts

```diff
@@ -17,27 +29,392 @@ export function generateRLCSamples(model: TestCodeModel, project: Project) {
+    if (!methods || methods.length == 0) {
+        return [];
+    }
+    const method = methods[0];
+    const hasInputParams = (!!rawMethodParams && rawMethodParams.length > 0), requireParam = !method.hasOptionalOptions;
+    if (!hasInputParams && !requireParam) {
+        return [];
+    }
+
+    const allSideAssignments = [], querySideAssignments: string[] = [], headerSideAssignments: string[] = [];
+    rawMethodParams.filter(p => p.parameter.protocol.http?.in == ParameterLocation.Body).forEach(p => {
+        allSideAssignments.push(` body: ` + getParameterAssignment(p.exampleValue));
+    });
+    rawMethodParams.filter(p => p.parameter.protocol.http?.in == ParameterLocation.Query).forEach(p => {
+        const name = getLanguageMetadata(p.parameter.language).serializedName || p.parameter.language.default.name;
+        querySideAssignments.push(`${name}: ` + getParameterAssignment(p.exampleValue));
+    });
+    if (querySideAssignments.length > 0) {
+        allSideAssignments.push(` queryParameters: { ` + querySideAssignments.join(", ") + `}`);
+    }
+    rawMethodParams.filter(p => p.parameter.protocol.http?.in == ParameterLocation.Header).forEach(p => {
+        const name = `"${getLanguageMetadata(p.parameter.language).serializedName}"`;
+        headerSideAssignments.push(`${name}: ` + getParameterAssignment(p.exampleValue));
+    });
+    if (headerSideAssignments.length > 0) {
+        allSideAssignments.push(` headers: { ` + headerSideAssignments.join(", ") + `}`);
+    }
+    let value: string = `{}`;
+    if (allSideAssignments.length > 0) {
+        value = `{ ` + allSideAssignments.join(", ") + `}`;
+    }
+    const optionParam: SampleParameter = {
+        name: "options",
+        assignment: `const options: ${method.optionsName} =` + value + `;`
+    }
+    addValueInImportedDict(getPackageName(), method.optionsName, importedDict);
+    return [optionParam];
+}
+
+function enrichParameterInSample(sample: RLCSampleDetail, parameters: SampleParameters) {
+    sample.clientParamAssignments = getAssignmentStrArray(parameters.client);
+    sample.clientParamNames = getContactParameterNames(parameters.client);
+    sample.pathParamAssignments = getAssignmentStrArray(parameters.path);
+    sample.pathParamNames = getContactParameterNames(parameters.path);
+    sample.methodParamAssignments = getAssignmentStrArray(parameters.method);
+    sample.methodParamNames = (parameters.method.length > 0) ? "options" : "";
+}
+
+function getAssignmentStrArray(parameters: SampleParameter[]) {
+    return parameters.filter(p => !!p.assignment).map(p => p.assignment!);
+}
+
+function getContactParameterNames(parameters: SampleParameter[]) {
+    return parameters.filter(p => p.name != null).map(p => p.name!).join(',');
 }
 
-function createSampleData(model: TestCodeModel) {
+function getParameterAssignment(exampleValue: ExampleValue) {
```

Comment:
This function looks familiar. Do we have a similar one in high level sample generation ? Can we reuse ?

Reply:
Yes, this is copied directly from HLC sample gen with slightly change for property name. Considering there is no any UnitTest could I do the refactoring in Phase 2 where I'll introduce more coner cases?